infrastructure/lib/table-permissions.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Supported Table Permissions for the application.
+ */
+export enum TablePermissions {
+  ALL = "ALL",
+  READ = "READ",
+  READ_WRITE = "READ_WRITE",
+  WRITE = "WRITE",
+}
```

Comment:
if you notice yourself creating an enum where some members are combinations of other members (like `READ_WRITE` or `ALL`) then you almost certainly would be better served by integers that you can `|`. So:

```ts
class TablePermissions {
  public const READ = 1 >> 0;
  public const WRITE = 1 >> 1;
  public const FOO = 1 >> 2;

  const get ALL() {
    return READ | WRITE | FOO;
  }
}


if (resource.permissions & TablePermissions.WRITE) {
  // resource.grantWriteAccess();
}
if (resource.permissions & TablePermissions.READ) {
  resource.grantReadAccess();
}
```